ReactAndroid/src/main/java/com/facebook/react/modules/network/ProgressResponseBody.java

```diff
@@ -0,0 +1,59 @@
+package com.facebook.react.modules.network;
+
+import java.io.IOException;
+
+import okhttp3.MediaType;
+import okhttp3.ResponseBody;
+import okio.Buffer;
+import okio.BufferedSource;
+import okio.ForwardingSource;
+import okio.Okio;
+import okio.Source;
+
+public class ProgressResponseBody extends ResponseBody {
+
+    private final ResponseBody mResponseBody;
+    private final ProgressListener mProgressListener;
+    private BufferedSource mBufferedSource;
+    private long mTotalBytesRead;
+
+    public ProgressResponseBody(ResponseBody responseBody, ProgressListener progressListener) {
```

Comment:
Ah yeah, guess we should mark it `@Nullable`.

I can amend the commit here on github, but it seems @davidaurelio is already amending the diff in FB land. If David doesn't add it while he's in there, I can send a trivial PR after this lands.